prov/efa/src/efa_fabric.c

```diff
@@ -63,16 +64,28 @@ static int efa_fabric_close(fid_t fid)
 static int efa_trywait(struct fid_fabric *fabric, struct fid **fids, int count)
 {
 	struct efa_cq *efa_cq;
+	struct util_cq *util_cq;
+	struct efa_rdm_cq *rdm_cq;
 	struct util_wait *wait;
 	int ret, i;
 
 	for (i = 0; i < count; i++) {
 		if (fids[i]->fclass == FI_CLASS_CQ) {
-			/* Use EFA-specific CQ trywait */
-			efa_cq = container_of(fids[i], struct efa_cq, util_cq.cq_fid.fid);
-			ret = efa_cq_trywait(efa_cq);
-			if (ret)
-				return ret;
+			util_cq = container_of(fids[i], struct util_cq, cq_fid.fid);
+			
+			/* RDM CQs use util wait objects, not hardware CQ events */
```

Comment:
what are the benefits of using util wait instead of efa device interrupt to implement wait? Is it because there can be too many device level completions that is only for 1 libfabric completion

Reply:
The major requirement of the blocking cq read is to avoid burning the CPU which was the busy loop that application will do when sread is not available. Are we confident this approach is meeting this requirement? cc @bwbarrett

Reply:
My understanding is we'd need SHM to expose some kind of wait object (e.g. `FI_WAIT_FD`) in order to implement any kind of multiplexing with EFA device interrupts. I'd be more than happy to engineer a solution but I figured that would be out of scope given the context/urgency of the feature. Could we consider this a follow-up/future improvement?

Short of that, don't we need to periodically drive progress or risk missing completions? I figured the exponential backoff strategy with a minimum interval of 1ms was appropriate since it matches backoff strategies we employ elsewhere. The `ofi_wait()` calls block on the util CQ FD between retries, so it's not "spinning," per se.

Reply:
>My understanding is we'd need SHM to expose some kind of wait object (e.g. FI_WAIT_FD) in order to implement any kind of multiplexing with EFA device interrupts. I'd be more than happy to engineer a solution but I figured that would be out of scope given the context/urgency of the feature. Could we consider this a follow-up/future improvement?

Considering the following scenario: when application have 2 threads, 1 thread poll the fd, 1 thread progress the cq via fi_read(NULL, 0). When we only have intra node (shm traffic), will thread 1 be signaled ? We know shm doesn't have any signals when writing the cq entries because it doesn't support wait